tests/components/myuplink/conftest.py

```diff
@@ -0,0 +1,110 @@
+"""Test helpers for myuplink."""
+from collections.abc import Generator
+import json
+import time
+from typing import Any
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from myuplink import Device, System
+import pytest
+
+from homeassistant.components.application_credentials import (
+    ClientCredential,
+    async_import_client_credential,
+)
+from homeassistant.components.myuplink.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .const import CLIENT_ID, CLIENT_SECRET
+
+from tests.common import MockConfigEntry, load_fixture, load_json_value_fixture
+
+
+@pytest.fixture(name="expires_at")
+def mock_expires_at() -> float:
+    """Fixture to set the oauth token expiration time."""
+    return time.time() + 3600
+
+
+@pytest.fixture
+def mock_config_entry(expires_at: int) -> MockConfigEntry:
+    """Return the default mocked config entry."""
+    return MockConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="myUplink test",
+        data={
+            "auth_implementation": DOMAIN,
+            "token": {
+                "access_token": "Fake_token",
+                "scope": "READSYSTEM offline",
+                "expires_in": 86399,
+                "refresh_token": "3012bc9f-7a65-4240-b817-9154ffdcc30f",
+                "token_type": "Bearer",
+                "expires_at": expires_at,
+            },
+        },
+        entry_id="myuplink_test",
+    )
+
+
+@pytest.fixture(autouse=True)
+async def setup_credentials(hass: HomeAssistant) -> None:
+    """Fixture to setup credentials."""
+    assert await async_setup_component(hass, "application_credentials", {})
+    await async_import_client_credential(
+        hass,
+        DOMAIN,
+        ClientCredential(
+            CLIENT_ID,
+            CLIENT_SECRET,
+        ),
+        DOMAIN,
+    )
+
+
+@pytest.fixture
+def mock_myuplink_client() -> Generator[AsyncMock, AsyncMock, AsyncMock]:
```

Comment:
```suggestion
def mock_myuplink_client() -> Generator[MagicMock, None, None]:
```

Reply:
Thanks, this was leftovers from my trialand error exercises.